app/src/main/java/com/example/poste/models/User.java

```diff
@@ -302,4 +304,62 @@ private String parseError(Response<ResponseBody> response) {
         }
         return null;
     }
+
+    /**
+     * Adds a new post the given folder
+     *
+     * @param folder folder to add the new post to
+     */
+    public void createNewPost(Folder folder, Context context, String title, String description, String url) {
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.createPost(getTokenHeader() ,new CreatePost(title, description, url, folder.getTitle()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                if (response.isSuccessful()) {
+                    updateFoldersAndPosts(context);
+                }
+                else {
+                    try {
+                        Toast.makeText(context, response.errorBody().string(), Toast.LENGTH_LONG);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+                Toast.makeText(context, "Server Not Responding", Toast.LENGTH_LONG);
+            }
+        });
+    }
+
+    /**
+     * Deletes a given post from a given folder in the data model and backend
+     *
+     * @param post the post you want to delete
+     */
+    public void deletePost(Context context, Post post, Folder folder) {
+        MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+        Call<ResponseBody> call = apiService.deletePost(getTokenHeader() ,new DeletePost(post.getId()));
+        call.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                if (response.isSuccessful()) {
+                    updateFoldersAndPosts(context);
+                }
+                else {
+                    try {
+                        Toast.makeText(context, response.errorBody().string(), Toast.LENGTH_LONG);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
```

Comment:
Careful with a RuntimeException. This immediately crashes the program. If unable to test this properly or match it up to the backend, I'd suggest a generic Toast message (and logging the error to console) to be fixed later.

Reply:
Switched to generic error toast.